DllImportGenerator/designs/StructMarshalling.md

```diff
@@ -41,6 +41,17 @@ public class MarshalUsingAttribute : Attribute
      public MarshalUsingAttribute(Type nativeType) {}
 }
 
+[AttributeUsage(AttributeTargets.Struct)]
+public class BlittableTypeIfGenericParametersBlittableAttribute : Attribute
+{
+     public BlittableTypeIfGenericParametersBlittableAttribute() {}
+}
+
+[AttributeUsage(AttributeTargets.GenericParameter)]
+public class ContributesToBlittabilityAttribute : Attribute
```

Comment:
An alternative to this would be to have the `BlittableTypeAttribute` type have an `int[]` property call `GenericTypesUsedInLayout`? This would reduce the proliferation of metadata cost on each generic type. I like the idea of targeting it but I also want to avoid attributes all over types if possible.